src/server/api/routers/user-level.ts

```
@@ -0,0 +1,196 @@
+};
+
+const addXp = async (userId: string, xpToAdd: number, db: DBSession) => {
+  const userLevel = await db.userLevel.findUnique({
+    where: { userId },
+  });
+
+  if (!userLevel) {
+    throwTRPCError("NOT_FOUND", "User level not found");
+  }
+
+  const currentXp = userLevel!.currentXp;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)